cmake/CommonStdlibBuiltins.cmake

```diff
@@ -0,0 +1,143 @@
+#
+#  Copyright (c) 2024, Intel Corporation
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+
+#
+# ispc CommonStdlibBuiltins.cmake
+#
+
+# Common function to determine architecture and OS
+function(determine_arch_and_os target bit os out_arch out_os)
+    set(arch "error")
+    if ("${target}" MATCHES "sse|avx")
+        if ("${bit}" STREQUAL "32")
+            set(arch "x86")
+        elseif ("${bit}" STREQUAL "64")
+            set(arch "x86_64")
+        else()
+            set(arch "error")
+        endif()
+    elseif ("${target}" MATCHES "neon")
+        if ("${bit}" STREQUAL "32")
+            set(arch "arm")
+        elseif ("${bit}" STREQUAL "64")
+            set(arch "aarch64")
+        else()
+            set(arch "error")
+        endif()
+    elseif ("${target}" MATCHES "wasm")
+        if ("${bit}" STREQUAL "32")
+            set(arch "wasm32")
+        elseif ("${bit}" STREQUAL "64")
+            set(arch "wasm64")
+        else()
+            set(arch "error")
+        endif()
+    elseif ("${target}" MATCHES "gen9|xe")
+        set(arch "xe64")
+    endif()
+
+    if ("${arch}" STREQUAL "error")
+        message(FATAL_ERROR "Incorrect target or bit passed: ${target} ${os} ${bit}")
+    endif()
+
+    if ("${os}" STREQUAL "unix")
+        set(os "linux")
+    endif()
+
+    set(${out_arch} ${arch} PARENT_SCOPE)
+    set(${out_os} ${os} PARENT_SCOPE)
+endfunction()
+
+# This function is a common entry to generate stdlib or target builtins via
+# stdlib_to_cpp or target_ll_to_cpp
+function (disp_target_stdlib func ispc_name target bit os CPP_LIST BC_LIST)
+    if (${func} STREQUAL "stdlib_to_cpp")
+        stdlib_to_cpp(${ispc_name} ${target} ${bit} ${os} ${CPP_LIST} ${BC_LIST})
+    elseif(${func} STREQUAL "target_ll_to_cpp")
+        target_ll_to_cpp(${target} ${bit} ${os} ${CPP_LIST} ${BC_LIST})
+    else()
+        message(FATAL_ERROR "Incorrect func name")
+    endif()
+
+    set(${CPP_LIST} ${${CPP_LIST}} PARENT_SCOPE)
+    set(${BC_LIST} ${${BC_LIST}} PARENT_SCOPE)
+endfunction()
+
+# Common function to call stdlib_to_cpp or target_ll_to_cpp. The actual name
+# is passed in func argument.
+function (generate_stdlib_or_target_builtins func ispc_name CPP_LIST BC_LIST)
+  message(STATUS "func ${func}")
```

Comment:
Looks like a debug output, which was forgotten to be removed.

Reply:
Fixed.